src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/applicationScopes.jsx

```diff
@@ -0,0 +1,98 @@
+import {Flex} from 'grid-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import Switch from 'app/components/switch';
+
+import {API_SCOPES} from 'app/constants';
+
+class ApplicationScopes extends React.Component {
+  static propTypes = {
+    scopes: PropTypes.array,
+    onToggle: PropTypes.func,
+  };
+
+  constructor(props) {
+    super(props);
+    let initialScopes = new Set(this.props.scopes);
+    this.state = {
+      loading: false,
+      error: false,
+      scopes: initialScopes,
+    };
+  }
+
+  handleToggleScopes = (scope, e) => {
+    let {scopes} = this.state;
```

Comment:
When we update state based on current state, we should use the `setState` callback form:


```javascript
this.setState(state => {
  ...
  return {
    scopes: new Set(scopes),
  };
});
```